packages/keyring-controller/src/KeyringController.ts

```diff
@@ -681,7 +690,7 @@ export class KeyringController extends BaseController<
    *
    * @returns The current state.
    */
-  async fullUpdate(): Promise<KeyringMemState> {
+  async fullUpdate(): Promise<KeyringControllerState> {
```

Comment:
This function doesn't return `KeyringControllerState` but `this.#keyring.fullUpdate()`, which is `EthKeyringController`'s memState. With `this.update` we should probably also update the vault, which is part of the persistent store; also, we should return both mem and persistent stores from the function.

Reply:
Persistence on both clients is handled through the store, rather than by inspecting the return values of these methods. So maybe we don't need to add the vault here (i.e. we could return `Omit(KeyringControllerState, 'vault'>` instead). Eventually I'd like to remove the state return values altogether.

That said.... I am curious about what this method is used for. It seems to be manually syncing the two state stores. Where is the vault synced? Could we use state change events instead, and get rid of this method altogether?

Reply:
I will refactor the controller to listen to `EthKeyringController`'s state updates to sync `KeyringController`'s store.

At the moment, I believe the vault is not being synced anywhere. Is it maybe that the responsible for persisting the vault is `EthKeyringController`? In this case, do we really need to have the vault in the `KeyringController`'s store?

Reply:
Uhm, my last suggestion of removing the vault from `KeyringController` might impact the reason why this migration to BaseControllerV2 is needed:

> We can consider updating core KeyringController to BaseControllerV2, which allows subscription on persistent and non-persistent data.

Reply:
Perhaps we can have the vault in `KeyringController` as non-persisted, and we sync it with a subscribe on `EthKeyringController` as we would do with the rest of the memStore

Reply:
This issue is being addressed in [this PR](https://github.com/MetaMask/core/pull/1384), so after its merge we can sync the vault also here (keeping it always non-persisted in `KeyringController`)